src/main/java/datawave/data/normalizer/NumberNormalizer.java

```diff
@@ -40,6 +40,10 @@ public String normalizeRegex(String fieldRegex) {
         }
     }
     
+    public ZeroRegexStatus getZeroRegexStatus(String untrimmedRegex) {
```

Comment:
Lets add a method to Type, or add another interface (like the OneToManyNormalizerType) with a method called something like requiresDenormalizedEvaluation().  I am sure we can come up with a better name however.

Reply:
I could see this also being used for LcNoDiacritics, but that may be too much of a change for users to accept.  There was a round of figuring out how to do case sensitive queries, and I believe that is why we created the #TEXT function.  We could have solved it this way instead maybe.  Oh well.

Reply:
> Lets add a method to Type, or add another interface (like the OneToManyNormalizerType) with a method called something like requiresDenormalizedEvaluation(). I am sure we can come up with a better name however.

I decided not to do it that way.  Whether a leading zeroes regex will requite denormed evaluation is dependent on the reverse index status of that field which I felt was improper for a Type to care about since we largely deal with index status back in the Visitors.

Reply:
Ah, that is a good point.  Contemplating....

Reply:
Since we already make decisions based on Type planning side, we can check if it's a regex node, if the field is a NumberType get the ZeroRegex status, add the denormalizedEvalOnly method there to check the field indexed/reverse indexed status, and make the determination about eval_only there.  That was the vision anyway.